compiler/ras/ILValidationUtils.cpp

```diff
@@ -0,0 +1,150 @@
+/*******************************************************************************
+ * Copyright (c) 2017, 2017 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *******************************************************************************/
+
+#include "ras/ILValidationUtils.hpp"
+
+#include <stdarg.h>                                   // for va_list
+
+#include "il/Node.hpp"                                // for Node
+#include "il/Node_inlines.hpp"                        // for Node::getChild, etc
+#include "il/ILProps.hpp"
+#include "il/ILOps.hpp"                               // for TR::ILOpCode
+
+
+TR::LiveNodeWindow::LiveNodeWindow(NodeSideTable<NodeState> &sideTable,
+                                   TR_Memory *memory)
+   :_sideTable(sideTable)
+   ,_basis(0)
+   ,_liveOffsets(10, memory, stackAlloc, growable)
+   {
+   }
+
+bool TR::isLoggingEnabled(TR::Compilation *comp)
+   {
+   return (comp->getOption(TR_TraceILValidation));
+   }
+
+void TR::checkCondition(TR::Node *node, bool condition,
+                        TR::Compilation *comp, const char *formatStr, ...)
+   {
+   if (!condition)
+      {
+      printDiagnostic(comp, "*** VALIDATION ERROR ***\nNode: %s n%dn\nMethod: %s\n",
+                      node->getOpCode().getName(), node->getGlobalIndex(),
+                      comp->signature());
+      va_list args;
+      va_start(args, formatStr);
+      vprintDiagnostic(comp, formatStr, args);
+      va_end(args);
+      printDiagnostic(comp, "\n");
+      /**
+       *Safely bring down the VM if the continueAfterILValidationError Compiler Option
+       *is not set.
+       */
+      if (!comp->getOption(TR_continueAfterILValidationError))
+         TR::trap();
+      }
+   }
+
+void TR::updateNodeState(TR::Node *node,
```

Comment:
I don't think it makes sense to use this function outside of `ValidateLivenessBoundaries`. Could you make it a helper function of that class instead?

Reply:
Good Call! Done.